Lighter color scheme and toggle button

colemanw opened the pull request:
Overview
--------

* Switches to a more modern looking white/grey color scheme.
* Style improvements for WP, D8, and D7.
* Adds a toggle button so users can quickly switch between "Replace CMS menu" and "Below CMS menu" positions.
* Adds a *none* option in display preferences.

Drupal 7 Example
-------
*Note improved interaction with the shortcuts bar*
![druapl7](https://user-images.githubusercontent.com/2874912/49197901-ca38de00-f35e-11e8-9250-40d55371fd48.gif)


Wordpress Example
-------
*Note interaction with the sidebar*
![wordpress](https://user-images.githubusercontent.com/2874912/49197920-dde44480-f35e-11e8-9d07-63788bf8bb9a.gif)

colemanw committed:
White is the new black

colemanw committed:
Add position toggle button to menubar

colemanw committed:
Override crm.drupal7.js

colemanw committed:
CMS style adjustments

christianwach commented:
@colemanw Wow, this is _so much better_ on WordPress now - lovely stuff! 👍 

Just to keep the feedback flowing, here are a few oddities I've noticed in no particular order:

### Fly-out sub-menu alignment

The fly-out sub-menus that are children of the final item in the primary dropdowns don't align properly:

<img width="364" alt="screen shot 2018-11-29 at 09 50 18" src="https://user-images.githubusercontent.com/726936/49213680-b8862380-f3bc-11e8-8de3-e0975402a009.png">

This is because of: 

```css
#civicrm-menu ul li:last-child a {
    margin-bottom: 5px;
}
```

Personally, I'd ditch this bottom margin (and the top margin too) as it's non-functional and therefore visually distracting. To me and my pixel-perfect pedantry, that is :-)

Speaking of which - the sub-arrows could do with a slight tweak to align better vertically:

```css
#civicrm-menu li li .sub-arrow:before {
    margin-top: 2px;
}
```

### Obscured elements

When the menu wraps (and it's likely to for many people - especially once a few extensions have added their top-level items), it obscures the UI elements below it:

<img width="893" alt="screen shot 2018-11-29 at 09 57 15" src="https://user-images.githubusercontent.com/726936/49213937-6396dd00-f3bd-11e8-8b7b-45fa4722a5eb.png">

I'd imagine this probably needs a Javascript solution, polling the height of the menu and setting the top padding of `#wp-body` as required. Not a problem when the menu is set to appear above the content area.

### Disable menu

* What is this setting for? 
* In what situations would I use it?

### Preparing for inclusion in core

How do I detect the presence of this menu? I currently check for the extension, but would appreciate knowing how this will pan out:

* Will this extension ship with CiviCRM and be auto-enabled?
* Will the code be ported to core and replace the old menu code?

Once again - thanks for your work on this Coleman; it's a great improvement.

aydun commented:
@vingle any comments from a Joomla perspective?

vingle commented:
> @vingle any comments from a Joomla perspective? From the description it does not affect Joomla

@aydun can't see anything that stands out skimming the commit but I need to test — I also agreed to test the revised menu on Joomla properly so will do together — am really hopeful that will be this week.

My one returning thought tho is: should we be left-aligning the items in the mobile menu? If I remember right it was only centered because we ran out of time at the sprint to make the mobile menu really nice?

colemanw committed:
Remove top/bottom margin that was throwing off submenu alignment

colemanw committed:
Adjust to menubar wrapping a new line & WP auto-fold

colemanw committed:
Fix height adjustments to only apply when menubar is visible

colemanw commented:
@christianwach really appreciate your flow of feedback - thanks for being so thorough. To your points:

1. *Fly-out sub-menu alignment* - I've taken your suggestion and removed the extra margin.
2. *Obscured elements* - I've added an extra body class `crm-menubar-wrapped` when the menubar is x2 height and added css adjustments.
3. *Disable menu* - I can't personally think of a use for it, but this is part of a general trend in which we migrate random php constants -> settings. Since I've already created this setting, seems like a perfect time to ditch the `CIVICRM_DISABLE_DEFAULT_MENU` constant as it was a simple matter of adding 1 more option.
4. *Preparing for inclusion in core* - This extension works with 5.8+ so anyone on the latest RC (soon to be the latest release) of CiviCRM can start using it today. My plan is to move this code into core ASAP (e.g. once #16 is complete) hopefully in 5.10, and yes it means I'll get to GUT the old menu code - what fun!

One more thing - I noticed that when the menubar was below the WP menu it wasn't responding to the `auto-fold` sidebar; so I added another breakpoint to fix that :)

colemanw commented:
I'm going to merge this as it includes fixes needed for testing.

aydun commented:
Re `CIVICRM_DISABLE_DEFAULT_MENU` - the early version of this extension used that.  Can't imagine why anyone would want to disable this lovely new version :-) ... but it seems reasonable to keep an option for another extension to do its own thing with menus.

christianwach commented:
Just quickly...

> ditch the `CIVICRM_DISABLE_DEFAULT_MENU` constant

@colemanw Please don't do that! 😟 At the moment, I'm counting on it being present in order to selectively disable the menu for certain users. Unless there's another way that I've missed - can I filter the setting for example?

colemanw commented:
Ok we can leave the constant in as well as the setting. This extension doesn't currently respect it but... hang on... 2d5f768
...now it does.

vingle commented:
> @vingle any comments from a Joomla perspective? From the description it does not affect Joomla

Have just tested. This looks great! Cdn't see any style bugs.

Stoob commented:
There's a fun post in the partners mattermost channel.  But seriously ... is there no way for a user to change the color without knowing how to override CSS?

christianwach commented:
@Stoob FWIW in WordPress CiviCRM Admin Utilities makes the menu match the colour scheme of the WordPress admin menu. You're welcome to [copy and modify the styles from there](https://github.com/christianwach/civicrm-admin-utilities/blob/master/assets/css/civicrm-admin-utilities-kam.css#L49-L101) if that helps.
